xchange-blockchain/src/main/java/org/knowm/xchange/blockchain/BlockchainAdapters.java

```diff
@@ -0,0 +1,256 @@
+            case "FILLED":
+                return Order.OrderStatus.FILLED;
+            case "EXPIRED":
+                return Order.OrderStatus.EXPIRED;
+            case "CANCELED":
+                return Order.OrderStatus.CANCELED;
+            case "REJECTED":
+                return Order.OrderStatus.REJECTED;
+            case "PENDING_CANCEL":
+                return Order.OrderStatus.PENDING_CANCEL;
+            case "PARTIALLY_FILLED":
+                return Order.OrderStatus.PARTIALLY_FILLED;
+            default:
+                return Order.OrderStatus.UNKNOWN;
+        }
+    }
+
+    public static OpenOrders toOpenOrders(List<BlockchainOrder> blockchainOrders){
+        List<LimitOrder> limitOrders = new ArrayList<>(Collections.emptyList());
+        List<Order> hiddenOrders = new ArrayList<>(Collections.emptyList());
+
+        for(BlockchainOrder blockchainOrder : blockchainOrders) {
+            final Order.OrderType orderType = blockchainOrder.isBuyer() ? Order.OrderType.BID : Order.OrderType.ASK;
+            final CurrencyPair symbol = blockchainOrder.getSymbol();
+            Order.Builder builder;
+
+            if (blockchainOrder.isMarketOrder()) {
+                builder = new MarketOrder.Builder(orderType, symbol);
+            } else if (blockchainOrder.isLimitOrder()){
+                builder = new LimitOrder.Builder(orderType, symbol).limitPrice(blockchainOrder.getPrice());
+            } else {
+                builder = new StopOrder.Builder(orderType, symbol).stopPrice(blockchainOrder.getPrice());
+            }
+            Order order = builder.orderStatus(toOrderStatus(blockchainOrder.getOrdStatus()))
+                    .originalAmount(blockchainOrder.getCumQty())
+                    .id(Long.toString(blockchainOrder.getExOrdId()))
+                    .timestamp(blockchainOrder.getTimestamp())
+                    .averagePrice(blockchainOrder.getAvgPx())
+                    .build();
+
+            if (order instanceof LimitOrder) {
+                limitOrders.add((LimitOrder) order);
+            } else {
+                hiddenOrders.add(order);
+            }
+        }
+
+        return new OpenOrders(limitOrders, hiddenOrders);
+    }
+
+    public static UserTrades toUserTrades(List<BlockchainOrder> blockchainTrades) {
+        List<UserTrade> trades = blockchainTrades.stream()
+                .map(blockchainTrade -> new UserTrade.Builder()
+                        .type(blockchainTrade.isBuyer()? Order.OrderType.BID : Order.OrderType.ASK)
+                        .originalAmount(blockchainTrade.getCumQty())
+                        .currencyPair(blockchainTrade.getSymbol())
+                        .price(blockchainTrade.getPrice())
+                        .timestamp(blockchainTrade.getTimestamp())
+                        .id(Long.toString(blockchainTrade.getExOrdId()))
+                        .orderId(blockchainTrade.getClOrdId())
+//                       .feeAmount(t.commission)
```

Comment:
why does this code is commented? We should remove it or implement it.

Reply:
This was implented for the poc, for the moment it will be deleted for this PR.